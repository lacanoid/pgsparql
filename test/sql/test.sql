select *
  from sparql.get_properties('dbpedia','http://dbpedia.org/resource/Johann_Sebastian_Bach')
 where predicate = 'http://dbpedia.org/ontology/birthYear'
;
select substr(sparql.compile_query('dbpedia','bach',$$
select ?predicate, ?object
where {
 <http://dbpedia.org/resource/Johann_Sebastian_Bach> ?predicate ?object.
}
$$,'{predicate}'),1,31) as magick
;
-- select * from bach
;
