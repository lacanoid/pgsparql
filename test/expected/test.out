select *
  from sparql.get_properties('dbpedia','http://dbpedia.org/resource/Johann_Sebastian_Bach')
 where predicate = 'http://dbpedia.org/ontology/birthYear'
;
               predicate               | object | value | lang 
---------------------------------------+--------+-------+------
 http://dbpedia.org/ontology/birthYear | 1685   | 1685  | 
(1 row)

select substr(sparql.compile_query('dbpedia','bach',$$
select ?predicate, ?object
where {
 <http://dbpedia.org/resource/Johann_Sebastian_Bach> ?predicate ?object.
}
$$,'{predicate}'),1,31) as magick
;
             magick              
---------------------------------
 create or replace function bach
(1 row)

-- select * from bach
;
